src/grep.rs

```diff
@@ -270,10 +279,11 @@ pub fn run(args: Args) -> io::Result<bool> {
 
   // Lock stdout before loop to prevent repetitive locking.
   let stdout = io::stdout();
-  let mut stdout = stdout.lock();
+  // let mut stdout = stdout.lock();
+  let stdout = &mut stdout.lock();
 
   // Reuse the same buffer for all the files, minimizing allocations.
-  let mut buffer = Vec::<u8>::new();
+  let buffer = &mut Vec::<u8>::new();
```

Comment:
I believe a reference has no benefit here. If I'm right, could you revert this please?